main.js

```diff
@@ -339,19 +369,79 @@ function checkWin(gameBoard, x, y) {
             countStonesInDirection(gameBoard, x, y, directions[i + 1][0], directions[i + 1][1], mostRecentStoneColor) - 1;
 
         if (total >= WINNING_NUMBER_OF_STONES) {
+            GAME_STATE.playerScores[mostRecentStoneColor] += 1;
+            GAME_STATE.scoreboardDiv.textContent = `Black: ${GAME_STATE.playerScores["black"]}  White: ${GAME_STATE.playerScores["white"]}`;
             return true;
         }
     }
 
     return false;
 }
 
+function undoGameHandlerFactory(gameBoard) {
+    return function (event) {
+        event.preventDefault();
+
+        if (GAME_STATE.stonesInScene.length > 0) {
+            let lastStone = GAME_STATE.stonesInScene.pop();
+            scene.remove(lastStone);
+
+            let x = Math.round(lastStone.position.x);
```

Comment:
I'm getting an error on this line.

I wonder what the rounding of positions is doing here? I would expect that in this "layer" we'd only be dealing with absolute coordinate positions, or even just loading a previous game state, seeing as replay seems to work perfecty.

Nice job on the replay, by the way!

Reply:
Thanks! The coordinates are a float. They get rounded so that the stones end up at the exact grid line crossings.